### DISCUSSION TITLE
Compile EGUI to Wasm Module

### DISCUSSION THREAD

**Discussion by felipetesc** [emilk/egui#2130] [Ideas]
Hello there! How can one proceed to compile all Egui library as wasm module to use it from JS/TS ? Accourdingly what was written here [:https://github.com/flyover/imgui-js ](https://developer.mozilla.org/en-US/docs/WebAssembly/Using_the_JavaScript_API) the first step is to create the wasm module, but I'm stuck about how to do that, can  someone gimme some ideas ?

PS: I know about egui_web at https://crates.io/crates/egui_web. That's not what I want. I want to use TS to create the UI

**Comment by emilk** [comment]
Look at https://github.com/emilk/eframe_template/

**Reply by felipetesc** [reply]
> Look at https://github.com/emilk/eframe_template/

Thanks for the quick reply. I've added a postscript, 'cause my post was slightly badly written. I want to use egui from TS, and not to write using rust, since TS is better for prototyping in my opinion.

**Reply by emilk** [reply]
If you want to use TypeScript to create a GUI, I think you need a GUI written for TS. egui is written for Rust, and there are no TS binding available or planned.

**Comment by felipetesc** [comment]
Yes, I was considering to create the bindings by myself. I just want some help to compile Egui for wasm. Before posting here I searched through the list of posts available, and I found this very interesting post https://github.com/emilk/egui/discussions/1324 . There is a link about imgui bindings for js. The project is this one: https://github.com/flyover/imgui-js Description summarizes:  JavaScript bindings for Dear ImGui using Emscripten and TypeScript

**Reply by felipetesc** [reply]
Oh. And even though, I want to use TS, if such thing can be done, we open the doors for Python, maybe Ruby, and who knows what other p. languages .... people will feel confortable to use ...
